compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionUtils.cpp

```diff
@@ -56,6 +63,118 @@ static bool canInsOperandTieWithOutsOperand(OpOperand *insOperand) {
   return llvm::any_of(linalgOp.getOutputOperands(), canTieWithOutsOperand);
 }
 
+/// Checks if a linalg op is a simple reduction of the innermost dimensions
+/// with identity map for the input.
+static bool isReductionOnInnermostDims(linalg::LinalgOp linalgOp) {
+  SmallVector<Operation *, 4> combinerOps;
+
+  auto numInputs = linalgOp.getNumInputs();
+  if (numInputs != 1 && numInputs != 2) return false;
+
+  if (linalgOp.getNumOutputs() != 1) return false;
+
+  // Check if the result dims are d0, d1, ..., which means the reduction is done
+  // in the innermost dimensions without an output transpose.
+  auto output = linalgOp.getOutputOperand(0);
+  auto outputIndexingMap = linalgOp.getTiedIndexingMap(output);
+  for (const auto &en : llvm::enumerate(outputIndexingMap.getResults())) {
+    auto expr = en.value();
+    if (auto dim = expr.dyn_cast<AffineDimExpr>()) {
+      if (dim.getPosition() != en.index()) return false;
+    } else {
+      return false;
+    }
+  }
+
+  if (!matchReduction(linalgOp.getRegionOutputArgs(), 0, combinerOps) ||
```

Comment:
We dont need to match the region of the operation to know it is a reduction. As long as the iterator_type says "reduction" that should be good enough.